### PR TITLE
tag-delta: Allow listing only downgrades or only upgrades

### PR DESCRIPTION
**Pull request by compi-migui**
Only really needed downgrades_only, but threw in the converse while I was at it.

### CODE REVIEW

File: tag_utils/cli/tag_delta.py [hunk 1]
```diff
@@ -69,39 +85,41 @@ def main():
...
+
+            if sys.stdout.isatty():
+                f = '%' + fw + 's %s%' + fw + 's %s%' + fw + 's'
+                print(f % (c, decor.HILIGHT, levr, decor.NORMAL, revr))
             else:
-                print(f'{c},{levr},{revr},')
+                if delta_info['downgrades'][c]['rebase']:
+                    print(f'{c},{levr},{revr},rebase')
+                else:
+                    print(f'{c},{levr},{revr},')
```
Comment (compi-migui) [inline tag_utils/cli/tag_delta.py:100]: The diff is having a hard time with this.  The block is indented one place right to be under the new `if show_downgrades:` conditional, there's no other changes.

File: tag_utils/cli/tag_delta.py [hunk 2]
```diff
@@ -69,39 +85,41 @@ def main():
...
-                print(f'{c},{levr},{revr},rebase')
+            if sys.stdout.isatty():
+                f = '%' + fw + 's %' + fw + 's %s%' + fw + 's%s'
+                print(f % (c, levr, decor.HILIGHT, revr, decor.NORMAL))
             else:
-                print(f'{c},{levr},{revr},')
+                if delta_info['upgrades'][c]['rebase']:
+                    print(f'{c},{levr},{revr},rebase')
+                else:
+                    print(f'{c},{levr},{revr},')
```
Comment (compi-migui) [inline tag_utils/cli/tag_delta.py:117]: Same as above:
The diff is having a hard time with this.  The block is indented one place right to be under the new `if show_upgrades:` conditional, there's no other changes.

**Review by jpichon — COMMENTED** [review]
LGTM! I'm wondering if it'd make sense to keep to the same terminology as --skip-same? So you'd have --skip-downgrades and --skip-upgrades as well, to avoid having to think in reverse depending on the flag. Then if for some reason, someone someday wants any combination of two options, they could also get it...

**Review by fuzzball81 — DISMISSED** [review]

**Comment by compi-migui** [comment]
> LGTM! I'm wondering if it'd make sense to keep to the same terminology as --skip-same? So you'd have --skip-downgrades and --skip-upgrades as well, to avoid having to think in reverse depending on the flag. Then if for some reason, someone someday wants any combination of two options, they could also get it...

hmm. For my use case this made much more sense, as I very often find myself wanting to list downgrades or upgrades and basically never need to see the intersection of the two tags. If anything (for how I use tag-delta) I'd get rid of --skip-same, making that behavior the default and have a --show-same option to include the intersection.

But it's totally fair that we'd want to keep the new options consistent with the existing one. I'll send up a new revision doing that.

Separately to that, if I have the time, I may propose reworking all the options in a way that makes them easier to use while keeping them internally consistent and hopefully also backwards-compatible. Need to think about that some more.

**Review by jpichon — APPROVED** [review]

**Review by jguiditta — APPROVED** [review]
lgtm

**Pull request merged**